MapboxCoreNavigation/RouteController.swift

```diff
@@ -122,30 +138,77 @@ open class RouteController: NSObject {
     
+    var isRerouting = false
     var lastRerouteLocation: CLLocation?
     
     var routeTask: URLSessionDataTask?
     
+    /// :nodoc: This is used internally when the navigation UI is being used
+    public var usesDefaultUserInterface = false
+    
+    var sessionState = SessionState()
+    var outstandingFeedbackEvents = [CoreFeedbackEvent]()
+    
     /**
      Intializes a new `RouteController`.
      
      - parameter route: The route to follow.
      - parameter directions: The Directions object that created `route`.
      - parameter locationManager: The associated location manager.
+     - parameter accessToken: Optional Mapbox accessToken.
      */
-    @objc(initWithRoute:directions:locationManager:)
-    public init(along route: Route, directions: Directions = Directions.shared, locationManager: NavigationLocationManager = NavigationLocationManager()) {
+    @objc(initWithRoute:directions:locationManager:accessToken:)
+    public init(along route: Route, directions: Directions = Directions.shared, locationManager: NavigationLocationManager = NavigationLocationManager(), accessToken: String? = nil) {
         self.directions = directions
         self.routeProgress = RouteProgress(route: route)
         self.locationManager = locationManager
         self.locationManager.activityType = route.routeOptions.activityType
         super.init()
         
         self.locationManager.delegate = self
+        self.resumeNotifications()
+        self.startEvents(accessToken: accessToken)
+        self.resetSession()
     }
     
     deinit {
         suspendLocationUpdates()
+        checkAndSendOutstandingFeedbackEvents(forceAll: true)
+        sendCancelEvent()
+        suspendNotifications()
+    }
+    
+    func startEvents(accessToken: String? = nil) {
+        var mapboxAccessToken: String? = nil
+        if let accessToken = accessToken {
+            mapboxAccessToken = accessToken
+        } else if let path = Bundle.main.path(forResource: "Info", ofType: "plist"),
+            let dict = NSDictionary(contentsOfFile: path) as? [String: AnyObject],
+            let token = dict["MGLMapboxAccessToken"] as? String {
+            mapboxAccessToken = token
+        }
+        
+        if let mapboxAccessToken = mapboxAccessToken {
+            events.isDebugLoggingEnabled = true
+            events.isMetricsEnabledInSimulator = true
+            events.isMetricsEnabledForInUsePermissions = true
+            events.initialize(withAccessToken: mapboxAccessToken, userAgentBase: "MapboxEventsNavigationiOS", hostSDKVersion: String(describing: Bundle(for: RouteController.self).object(forInfoDictionaryKey: "CFBundleShortVersionString")!))
+            events.disableLocationMetrics()
+            events.sendTurnstileEvent()
+        } else {
+            assert(false, "`accessToken` must be set in the Info.plist as `MGLMapboxAccessToken`m passed in as a String to `RouteController()`")
```

Comment:
To close the loop: this was unblocked upstream in mapbox/MapboxDirections.swift#155.